src/DependencyInjection/SafeMigrationsExtension.php

```diff
@@ -4,15 +4,64 @@
 
 namespace Eniams\SafeMigrationsBundle\DependencyInjection;
 
+use Eniams\SafeMigrationsBundle\MigrationFileSystem;
+use Eniams\SafeMigrationsBundle\Statement\StatementInterface;
+use Eniams\SafeMigrationsBundle\Warning\WarningFactory;
+use Symfony\Component\Config\FileLocator;
 use Symfony\Component\DependencyInjection\ContainerBuilder;
-use Symfony\Component\DependencyInjection\Extension\Extension;
+use Symfony\Component\DependencyInjection\Definition;
 
-final class SafeMigrationsExtension extends Extension
+use function Symfony\Component\DependencyInjection\Loader\Configurator\tagged_iterator;
+
+use Symfony\Component\DependencyInjection\Loader\PhpFileLoader;
+use Symfony\Component\HttpKernel\DependencyInjection\ConfigurableExtension;
+
+final class SafeMigrationsExtension extends ConfigurableExtension
 {
-    public function load(array $configs, ContainerBuilder $container): void
+    public function loadInternal(array $mergedConfig, ContainerBuilder $container): void
+    {
+        $loader = new PhpFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
+        $loader->load('services.php');
+
+        $migrationsPath = $mergedConfig['migrations_path'];
+        $criticalTables = $mergedConfig['critical_tables'];
+        $excludedStatements = $mergedConfig['excluded_statements'];
+
+        if (!is_dir($migrationsPath)) {
+            throw new \InvalidArgumentException(sprintf('You must provide a valid path to your migrations directory. "%s" given.', $migrationsPath));
+        }
+
+        $container->setDefinition('eniams.safe_migrations.file_system', new Definition(MigrationFileSystem::class))
+            ->setArgument(0, $migrationsPath)
+        ;
+
+        $container->registerForAutoconfiguration(StatementInterface::class)->addTag('eniams.safe_migrations.statement');
+        //        $excludedServiceStatements = $this->getExcludedStatements($container, $excludedStatements);
+
+        $container->setDefinition('eniams.safe_migrations.warning_factory', new Definition(WarningFactory::class))
+            ->setArguments([
+                tagged_iterator('eniams.safe_migrations.statement'),
+                $criticalTables,
+                $excludedStatements,
```

Comment:
@ismail1432 it works when adding "raw statements" instead of service's ids. But it is the correct way to to this?